spec/PhpSpec/Locator/BestMatchResourceManagerSpec.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace spec\PhpSpec\Locator;
+
+use PhpSpec\Exception\Locator\ResourceCreationException;
+use PhpSpec\Locator\Resource;
+use PhpSpec\Locator\ResourceLocator;
+use PhpSpec\ObjectBehavior;
+
+class BestMatchResourceManagerSpec extends ObjectBehavior
+{
+    function let(ResourceLocator $locator1, ResourceLocator $locator2)
+    {
+        $locator1->getPriority()->willReturn(5);
+        $locator2->getPriority()->willReturn(10);
+    }
+
+    function it_locates_resources_using_all_registered_locators($locator1, $locator2,
+        Resource $resource1, Resource $resource2, Resource $resource3
+    ) {
+        $this->registerLocator($locator1);
+        $this->registerLocator($locator2);
+
+        $locator1->supportsQuery('s:query')->willReturn(true);
+        $locator1->findResources('s:query')->willReturn(array($resource3, $resource2));
+        $locator2->supportsQuery('s:query')->willReturn(true);
+        $locator2->findResources('s:query')->willReturn(array($resource1));
+
+        $resource1->getSpecClassname()->willReturn('Some\Spec1');
+        $resource2->getSpecClassname()->willReturn('Some\Spec2');
+        $resource3->getSpecClassname()->willReturn('Some\Spec3');
+
+        $this->locateResources('s:query')->shouldReturn(array($resource1, $resource3, $resource2));
+    }
+
+    function it_locates_all_locators_resources_if_query_string_is_empty($locator1, $locator2,
+                                                                        Resource $resource1, Resource $resource2, Resource $resource3
```

Comment:
something is messed up with whitespaces here, could you make it more PSR-2 compliant, please?